util/helpfulIO.py

```diff
@@ -0,0 +1,120 @@
+from typing import Callable
+from ctre import WPI_TalonFX, ControlMode
+from wpilib import RobotBase, PWMVictorSPX, DigitalInput
+from wpimath.controller import PIDController
+from wpimath.system.plant import DCMotor
+from util.ctrecheck import ctreCheckError
+
+import constants
+
+LimitSwitch = Callable[[], bool]
+
+
+class Falcon:  # represents either a simulated motor or a real Falcon 500
+    def __init__(
+        self,
+        name: str,
+        realId: int,
+        simId: int,
+        PGain: int = 1,
+        IGain: int = 0,
+        DGain: int = 0,
+        PIDSlot: int = 0,
+    ) -> None:  # depending on if its in simulation or the real motor, different motors will be used
+        self.name = name
+        self.motor = WPI_TalonFX(realId) if RobotBase.isReal() else PWMVictorSPX(simId)
```

Comment:
I can't find a spot where you use the PWMVictorSPX in the code, since you instead use DCMotor.falcon500. Do we still need that?

Reply:
its the defacto for if its not a real motor, the falcon500 is simply for getting the amount given speed

Reply:
Is it used anywhere though? It seems self.motor is only used when the robot isReal